app/services/dish_search_service.rb

```diff
@@ -0,0 +1,86 @@
+class DishSearchService
```

Comment:
Consider dropping the `Service` suffix from the name if you can come up with a name that describes the class's responsibility more succinctly and accurately without it. This can help avoid an app having an eventual excess of classes with a name ending `Service`.

Here `DishSearcher` or `DishSearch` would work.